libfoundation: Assume stdint.h and stddef.h are always present

peter-b opened the pull request:
Now that we are compiling LiveCode in C++11 mode, the two standard C
headers "stdint.h" and "stddef.h" are available on all the platforms
we compile for.

This allows the removal of lots of preprocessor magic from the
"foundation.h" header.

peter-b committed:
libfoundation: Assume stdint.h and stddef.h are always present

Now that we are compiling LiveCode in C++11 mode, the two standard C
headers "stdint.h" and "stddef.h" are available on all the platforms
we compile for.

This allows the removal of lots of preprocessor magic from the
"foundation.h" header.

livecodefraser commented:
@livecode-vulcan review okay 98d2322

livecode-vulcan commented:
💙 review by @livecodefraser ok 98d2322364abba0ac05e6e4683c0f15b148a5976

livecode-vulcan commented:
😞 test failure 98d2322364abba0ac05e6e4683c0f15b148a5976
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2715)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2725)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2710)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2726)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2732)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2738)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2750)

livecodefraser commented:
Ah, the lack of ssize_t is causing problems.

peter-b committed:
libfoundation: Ensure ssize_t is defined on Windows.

Unfortunately Windows 7.0A SDK doesn't provide a usable definition of
`ssize_t`.

livecodefraser commented:
@livecode-vulcan review okay cae11c1

livecode-vulcan commented:
💙 review by @livecodefraser ok cae11c16e8a498a0ef841293560dc8968d175eb5

livecode-vulcan commented:
😞 test failure cae11c16e8a498a0ef841293560dc8968d175eb5
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2717)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2727)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2712)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2728)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2734)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2740)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2752)

peter-b commented:
This isn't going to work because the QuickTime SDK, which we still use on Windows, is inserting its own incorrect version of `stdint.h` into the Visual Studio compiler's header search path.

peter-b commented:
@livecode-vulcan retry cae11c1

livecode-vulcan commented:
😞 test failure cae11c16e8a498a0ef841293560dc8968d175eb5
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2720)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2730)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2715)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2731)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2737)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2743)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2755)

livecodefraser commented:
@livecode-vulcan review okay 661a10c

livecode-vulcan commented:
💙 review by @livecodefraser ok 661a10ccb398814c9fcef539e44c0496142a836d

livecode-vulcan commented:
😎 test success 661a10ccb398814c9fcef539e44c0496142a836d
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/2721)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/2731)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/2716)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/2732)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/2738)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/2744)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/2756)
